apps/flame-defi/app/pool/context/pool-position-context.tsx

```diff
@@ -1,132 +1,264 @@
+    setFeeTier(feeTier);
+  };
+
+  const getPriceRange = useCallback(async () => {
+    if (!address) {
+      return;
+    }
+
+    try {
+      const position = await nonFungiblePositionService.positions(
+        selectedChain.chainId,
+        tokenId,
+      );
+      const factoryService = createPoolFactoryService(
+        wagmiConfig,
+        selectedChain.contracts.poolFactory.address,
+      );
+
+      const token0 = getTokenDataFromCurrencies(
+        currencies,
+        position.tokenAddress0,
+      );
+
+      const token1 = getTokenDataFromCurrencies(
+        currencies,
+        position.tokenAddress1,
+      );
+
+      const poolAddress = await factoryService.getPool(
+        selectedChain.chainId,
+        position.tokenAddress0,
+        position.tokenAddress1,
+        position.fee,
+      );
+      const poolService = createPoolService(wagmiConfig, poolAddress);
+      const slot0 = await poolService.getSlot0(selectedChain.chainId);
+      const pricePerToken = sqrtPriceX96ToPrice(
+        slot0.sqrtPriceX96,
+        invertedPrice,
+        token0?.coinDecimals,
+        token1?.coinDecimals,
+      );
+      setCurrentPrice(pricePerToken.toFixed(6));
+
+      if (position) {
+        // Get the correct min/max tick values based on fee tier
+        const { MIN_TICK, MAX_TICK } = getMinMaxTick(position.fee);
+
+        // Check if position spans the full range
+        const isFullRange =
+          position.tickLower === MIN_TICK && position.tickUpper === MAX_TICK;
+
+        if (isFullRange) {
+          setMinPrice("0");
+          setMaxPrice("∞");
+        } else {
+          const minPriceFromTick = tickToPrice(
+            position.tickLower,
+            token0?.coinDecimals,
+            token1?.coinDecimals,
+          );
+          const maxPriceFromTick = tickToPrice(
+            position.tickUpper,
+            token0?.coinDecimals,
+            token1?.coinDecimals,
+          );
+          setMinPrice(minPriceFromTick.toFixed(6));
+          setMaxPrice(maxPriceFromTick.toFixed(6));
+        }
+      }
+    } catch (error) {
+      console.error("Error fetching price range:", error);
+    }
+  }, [
+    address,
+    nonFungiblePositionService,
+    selectedChain,
+    tokenId,
+    currencies,
+    wagmiConfig,
+    invertedPrice,
+  ]);
+
+  if (poolTokens.length === 0) {
+    getPoolTokens();
+    getFeeTier();
+  }
+
+  useEffect(() => {
+    getPriceRange();
+  }, [invertedPrice, getPriceRange]);
 
   const handleCollectAsNative = (collectAsNative: boolean) => {
     setCollectAsNative(collectAsNative);
     setInLocalStorage("poolSettings", {
       collectAsNative: collectAsNative,
     });
     if (collectAsNative) {
-      positionDetails?.tokens.map((token) => {
+      poolTokens?.map((token) => {
         if (token.symbol === "TIA") {
           token.symbol = "WTIA";
         }
       });
     } else {
-      positionDetails?.tokens.map((token) => {
+      poolTokens?.map((token) => {
```

Comment:
The `?` is not needed here. `poolTokens` might be `[]` but it will never be `null` or `undefined`